File: devbricksx-java-compiler/src/main/java/com/dailystudio/devbricksx/compiler/utils/NameUtils.java
package com.dailystudio.devbricksx.compiler.utils;

public class NameUtils {

    public static String capitalizeName(String name) {
        if (TextUtils.isEmpty(name)) {
            return null;
        }

        return name.substring(0, 1).toUpperCase() + name.substring(1);
    }

    public static String lowerCamelCaseName(String name) {
        if (TextUtils.isEmpty(name)) {
            return null;
        }

        StringBuilder builder = new StringBuilder(name);

        for (int i = 0; i < name.length(); i++) {
            if (Character.isAlphabetic(name.charAt(i))) {
                builder.setCharAt(i, Character.toLowerCase(name.charAt(i)));
                break;
            }
        }

        return builder.toString();
    }

    public static String underscoreCaseName(String name) {
        if (TextUtils.isEmpty(name)) {
            return null;
        }

        StringBuilder builder = new StringBuilder();
        char ch;
        for (int i = 0; i < name.length(); i++) {
            ch = name.charAt(i);
            if (!Character.isAlphabetic(ch)
                    || Character.isLowerCase(ch)) {
                builder.append(ch);
            } else {
                builder.append('_');
                builder.append(Character.toLowerCase(ch));
            }
        }

        return builder.toString();
    }

}

File: android/devbricksx-kotlin-compiler/src/main/java/com/dailystudio/devbricksx/compiler/kotlin/utils/NameUtils.kt
package com.dailystudio.devbricksx.compiler.kotlin.utils

import java.lang.StringBuilder

fun String.lowerCamelCaseName(): String {
    if (isBlank()) {
        return this
    }

    val builder = StringBuilder(this)

    for (i in this.indices) {
        if (Character.isAlphabetic(this[i].toInt())) {
            builder.setCharAt(i, Character.toLowerCase(this[i]))
            break
        }
    }

    return builder.toString()
}